src/Kovan-DssSpell.sol

```diff
@@ -15,71 +15,160 @@
+        // Authorize the new END in contracts
+        DssExecLib.authorize(MCD_VAT, MCD_END);
+        DssExecLib.authorize(MCD_CAT, MCD_END);
+        DssExecLib.authorize(MCD_DOG, MCD_END);
+        DssExecLib.authorize(MCD_VOW, MCD_END);
+        DssExecLib.authorize(MCD_POT, MCD_END);
+        DssExecLib.authorize(MCD_SPOT, MCD_END);
 
-    address constant PROXY_ACTIONS = 0xD8b9702755E91Aa792656966aE6bAF32F4C394Ba;
+        // Deauthorize the old END in contracts
+        DssExecLib.deauthorize(MCD_VAT, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_CAT, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_VOW, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_POT, MCD_END_OLD);
+        DssExecLib.deauthorize(MCD_SPOT, MCD_END_OLD);
 
-    function actions() public override {
-        CollateralOpts memory PAXG_A = CollateralOpts({
-            ilk: "PAXG-A",
-            gem: PAXG,
-            join: MCD_JOIN_PAXG_A,
-            flip: MCD_FLIP_PAXG_A,
-            pip: PIP_PAXG,
-            isLiquidatable: true,
-            isOSM: true,
-            whitelistOSM: true,
-            ilkDebtCeiling: 5 * MILLION,
-            minVaultAmount: 100,
-            maxLiquidationAmount: 500,
-            liquidationPenalty: 1300,
-            ilkStabilityFee: FOUR_PCT,
-            bidIncrease: 300,
-            bidDuration: 1 hours,
-            auctionDuration: 1 hours,
-            liquidationRatio: 12500
-        });
-        addNewCollateral(PAXG_A);
-
-        DssExecLib.setChangelogAddress("PAXG", PAXG);
-        DssExecLib.setChangelogAddress("MCD_JOIN_PAXG_A", MCD_JOIN_PAXG_A);
-        DssExecLib.setChangelogAddress("MCD_FLIP_PAXG_A", MCD_FLIP_PAXG_A);
-        DssExecLib.setChangelogAddress("PIP_PAXG", PIP_PAXG);
-
-        DssExecLib.setChangelogAddress("PROXY_ACTIONS", PROXY_ACTIONS);
-
-        // Turn on Faucet
-        FaucetAbstract(DssExecLib.getChangelogAddress("FAUCET")).setAmt(PAXG, WAD);
+        // Authorize new ESM to execute in new END
+        DssExecLib.authorize(MCD_END, MCD_ESM);
+
+        // Deuthorize old ESM to execute in old END
+        DssExecLib.deauthorize(MCD_END_OLD, MCD_ESM_OLD);
+
+        // ------------------  AUCTION  ------------------
+
+        ClipperMomLike(CLIPPER_MOM).setAuthority(DssExecLib.getChangelogAddress("MCD_ADM"));
+
+        // Set VOW in the DOG
+        DssExecLib.setContract(MCD_DOG, "vow", MCD_VOW);
+
+        // Set CLIP for LINK-A in the DOG
+        DssExecLib.setContract(MCD_DOG, "LINK-A", "clip", MCD_CLIP_LINK_A);
+
+        // Set VOW in the LINK-A CLIP
+        DssExecLib.setContract(MCD_CLIP_LINK_A, "vow", MCD_VOW);
+
+        // Set CALC in the LINK-A CLIP
+        DssExecLib.setContract(MCD_CLIP_LINK_A, "calc", MCD_CLIP_CALC_LINK_A);
+
+        // Authorize CLIP can access to VAT
+        DssExecLib.authorize(MCD_VAT, MCD_CLIP_LINK_A);
+
+        // Authorize CLIP can access to DOG
+        DssExecLib.authorize(MCD_DOG, MCD_CLIP_LINK_A);
+
```

Comment:
I think we can authorize the ESM on clippers moving forward to prevent the race condition where governance can steal collateral. This works with the PR: https://github.com/makerdao/esm/pull/20

Reply:
Yes I have that already done in my local.

Reply:
This was pushed